src/RoomsSDKAdapter.js

```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
       );
 
       // Convert to a multicast observable
-      this.getRoomObservables[ID] = getRoom$.pipe(
-        publishReplay(1),
-        refCount(),
-      );
+      this.getRoomObservables[ID] = getRoom$.pipe(publishReplay(1), refCount());
     }
 
     return this.getRoomObservables[ID];
   }
+
+  /**
+   * Returns an observable that emits current and future activities from the specified room.
+   *
+   * @param {string} id ID of room to get
+   * @returns {Observable.<Activity>} Observable stream that emits current and future activities from the specified room
+   */
+  getRoomActivities(id) {
+    logger.debug('ROOM', id, 'getRoomActivities()', ['called with', {id}]);
+
+    // if (!(ID in this.getRoomObservables)) {
+    //   return throwError(new Error(`Room ${ID} does not exist`));
+    // }
+
+    const activity$ = new BehaviorSubject(null);
+
+    this.datasource.internal.mercury.on(
+      'event:conversation.activity',
+      (activity) => {
+        const {id: hydraId} = deconstructHydraId(id);
+
+        if (activity.target && activity.target.id === hydraId) {
+          const activityInterface = {
```

Comment:
This is not really an interface at this point, so this variable name is misleading. This is more like an activity of the component system
```suggestion
          const componentActivity = {
```

Reply:
Alternatively, you can just simply call this "activity" and the one from the SDK "sdkActivity". This is the pattern we follow in other modules
```suggestion
          const activity = {
```